### PR TITLE
Refactor Brave Ads in preparation for Publisher Ads

### PR DESCRIPTION
**Pull request by tmancey**
Resolves https://github.com/brave/brave-browser/issues/8445

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:

- Confirm ad notifications are shown and the user is rewarded

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: components/brave_ads/browser/ads_service_impl.cc [hunk 1]
```diff
@@ -1967,15 +1967,16 @@ void AdsServiceImpl::SetCatalogIssuers(
...
+    const ads::AdInfo& info,
+    const ads::ConfirmationType confirmation_type) {
+  rewards_service_->ConfirmAd(info.ToJson(), confirmation_type);
 }
 
 void AdsServiceImpl::ConfirmAction(
     const std::string& creative_instance_id,
     const std::string& creative_set_id,
-    const ads::ConfirmationType& confirmation_type) {
+    const ads::ConfirmationType confirmation_type) {
```
Comment (masparrow) [inline components/brave_ads/browser/ads_service_impl.cc]: Just checking - absolutely sure you want to pass object copies?

File: vendor/bat-native-ads/include/bat/ads/ad_content.h [hunk 2]
```diff
@@ -45,7 +45,7 @@ struct ADS_EXPORT AdContent {
   std::string brand_display_url;
   std::string brand_url;
   LikeAction like_action = LikeAction::kNone;
-  ConfirmationType ad_action = ConfirmationType::kUnknown;
+  ConfirmationType ad_action;
```
Comment (masparrow) [inline vendor/bat-native-ads/include/bat/ads/ad_content.h]: Removing default init value? Curious as to why?

File: components/brave_ads/browser/ads_service_impl.cc (hunk 1 above)
Comment (tmancey) [reply components/brave_ads/browser/ads_service_impl.cc]: ConfirmationType is an enum/int

File: vendor/bat-native-ads/include/bat/ads/ad_content.h (hunk 2 above)
Comment (tmancey) [reply vendor/bat-native-ads/include/bat/ads/ad_content.h]: Removed unknown type as no longer required

**Comment by tmancey** [comment]
CI failed for unrelated tests. Those tests pass locally on my machine. Merging with master

**Pull request merged**